### PR TITLE
[polaris.shopify.com] Add link to component lifecycle in alpha banner

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Link to the [component lifecycle page](https://polaris.shopify.com/getting-started/components-lifecycle) from our alpha and beta component warning so people can learn about alpha status.

Screenshot 2023-01-31 at 12 01 08 PM

Screenshot 2023-01-31 at 12 13 36 PM

### WHAT is this pull request doing?
replacing the `p` in the status banner component and instead rendering the `message` prop with the `Markdown` component so that we can add links to the `message`

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/src/components/StatusBanner/StatusBanner.tsx [hunk 1]
```diff
@@ -10,7 +12,7 @@ function StatusBanner({status: {value, message}}: Props) {
   return (
     <div className={styles.StatusBanner} data-value={value.toLowerCase()}>
       <h2>{uppercaseFirst(value)}</h2>
-      <p>{message}</p>
+      <Markdown text={message} />
```
Comment (yurm04) [inline polaris.shopify.com/src/components/StatusBanner/StatusBanner.tsx:14]: any reason we wouldn't want to support markdown here in the future?

File: polaris.shopify.com/content/components/alpha-stack.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ keywords:
   - layout
 status:
   value: Alpha
-  message: This component and its documentation is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution.
+  message: This component and its documentation is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution. To learn more please read our [Component lifecycle resource](/getting-started/components-lifecycle).
```
Comment (yurm04) [inline polaris.shopify.com/content/components/alpha-stack.md]: we cool with this message?

File: polaris.shopify.com/content/components/alpha-stack.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ keywords:
   - layout
 status:
   value: Alpha
-  message: This component and its documentation is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution.
+  message: This component and its documentation is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution. To learn more please read about our [component lifecycle](/getting-started/components-lifecycle).
```
Comment (yurm04) [inline polaris.shopify.com/content/components/alpha-stack.md]: we cool with this message?

Comment (sarahill) [reply polaris.shopify.com/content/components/alpha-stack.md]:
```suggestion
  message: This component is a work in progress and ready for exploratory usage, with breaking changes expected in minor version updates. Please use with caution. Learn more about our [component lifecycle](/getting-started/components-lifecycle).
```

File: polaris.shopify.com/content/components/text.md [hunk 4]
```diff
@@ -21,7 +21,7 @@ keywords:
   - list
 status:
   value: Beta
-  message: This component is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution.
+  message: This component is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution. To learn more please read about our [component lifecycle](/getting-started/components-lifecycle).
```
Comment (aveline) [inline polaris.shopify.com/content/components/text.md]:
```suggestion
  message: This component is ready for wider adoption, usage is encouraged for most cases. Breaking changes are possible in minor version updates. To learn more please read about our [component lifecycle](/getting-started/components-lifecycle).
```

**Review by aveline — CHANGES_REQUESTED** [review]

File: polaris.shopify.com/content/components/alpha-card.md [hunk 5]
```diff
@@ -22,7 +22,7 @@ keywords:
   - call out
 status:
   value: Alpha
-  message: This component and its documentation is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution.
+  message: This component and its documentation is in development. There could be breaking changes made to it in a non-major release of Polaris. Please use with caution. To learn more please read about our [components lifecycle](/getting-started/components-lifecycle).
```
Comment (sarahill) [inline polaris.shopify.com/content/components/alpha-card.md]: One more suggestion to this copy to match the lifecycle docs.

```suggestion
  message: This component is a work in progress and ready for exploratory usage, with breaking changes expected in minor version updates. Please use with caution. Learn more about our [components lifecycle](/getting-started/components-lifecycle).
```

**Review by sarahill — APPROVED** [review]
Left a couple copy comments but overall looks good! Suggested a change to "Learn more" formatting in one place but if we go with the suggestion let's update it across these banners.

Thanks for updating this 🎉

**Review by aveline — APPROVED** [review]

**Review by chazdean — APPROVED** [review]

**Review by alex-page — APPROVED** [review]

**Comment by alex-page** [comment]
@yurm04 could this sentence be added to the banner component as a string or with a property. Then it doesn't have to be added with content in each markdown file?

File: polaris.shopify.com/src/components/StatusBanner/StatusBanner.tsx (hunk 1 above)
Comment (alex-page) [reply polaris.shopify.com/src/components/StatusBanner/StatusBanner.tsx:14]: Nope.

**Comment by yurm04** [comment]
@alex-page i assumed that we still wanted to be able to override the banner with a custom message. would something like this be what you're thinking about?

```jsx
// value is alpha, beta, warning, etc for the banner styling
// if message is empty then render a default message based on value?
function StatusBanner({status: {value, message}}: Props) {
  if (!message) {
    switch (value) {
      case 'alpha':
        message = 'This component is a work in progress and ready for exploratory usage, with breaking changes expected in minor version updates. Please use with caution. Learn more about our Component lifecycle](/getting-started/components-lifecycle).'
        break;
    
      case 'beta':
        message = 'This component is ready for wider adoption, usage is encouraged for most cases. Breaking changes are possible in minor version updates. To learn more please read about our [Component lifecycle](/getting-started/components-lifecycle)'
        break;
      default:
        break;
    }
  }
  return (
    <div className={styles.StatusBanner} data-value={value.toLowerCase()}>
      <h2>{uppercaseFirst(value)}</h2>
      <Markdown text={message} />
    </div>
  );
}
```

**Review by laurkim — APPROVED** [review]
Nice 💯 
Thank you for updating the banners!

**Review by martenbjork — APPROVED** [review]
🚀

**Pull request merged**